pkg/kubectl/cmd/run.go

```diff
@@ -304,45 +350,98 @@ func contains(resourcesList map[string]*unversioned.APIResourceList, resource un
 	return false
 }
 
-func waitForPodRunning(c *client.Client, pod *api.Pod, out io.Writer, quiet bool) (status api.PodPhase, err error) {
+func waitForPod(c *client.Client, ns, name string, exitCondition func(*api.Pod) bool, tick func(*api.Pod)) (*api.Pod, error) {
```

Comment:
`tick` comes from the Go `ticker`: https://golang.org/pkg/time/#Ticker. It's called on each tick of the timer.